pkg/minikube/openshiftversions/openshift_versions.go

```diff
@@ -17,48 +17,125 @@ limitations under the License.
+	Creator     int         `json:"creator"`
+	LastUpdater int         `json:"last_updater"`
+	LastUpdated time.Time   `json:"last_updated"`
+	ImageID     interface{} `json:"image_id"`
+	V2          bool        `json:"v2"`
+}
 
-func PrintOpenShiftVersionsFromGitHub(output io.Writer) {
-	PrintOpenShiftVersions(output)
+type ImageInfo struct {
+	Size         int         `json:"size"`
+	Architecture string      `json:"architecture"`
+	Variant      interface{} `json:"variant"`
+	Features     interface{} `json:"features"`
+	Os           interface{} `json:"os"`
+	OsVersion    interface{} `json:"os_version"`
+	OsFeatures   interface{} `json:"os_features"`
 }
 
-func PrintOpenShiftVersions(output io.Writer) {
-	versions, err := getVersions()
+func PrintDownStreamVersions(output io.Writer, minSupportedVersion string) {
+	resp, err := getResponseBody("https://registry.access.redhat.com/v1/repositories/openshift3/ose/tags")
 	if err != nil {
-		fmt.Println(err.Error())
+		fmt.Println("Error Occured", err)
 		return
 	}
-	fmt.Fprint(output, "The following OpenShift versions are available: \n")
-
-	for _, version := range versions {
-		if util.ValidateOpenshiftMinVersion(*version.TagName) {
-			fmt.Fprintf(output, "\t- %s\n", *version.TagName)
+	defer resp.Body.Close()
+	decoder := json.NewDecoder(resp.Body)
+	var data map[string]string
+	err = decoder.Decode(&data)
+	if err != nil {
+		fmt.Printf("%T\n%s\n%#v\n", err, err, err)
+		return
+	}
+	fmt.Fprintf(output, "The following OpenShift versions are available: \n")
+	for version := range data {
+		if versionOrdinal(version) > versionOrdinal(minSupportedVersion) {
+			if strings.Contains(version, "-") {
+				continue
+			}
+			fmt.Fprintf(output, "\t- %s\n", version)
 		}
 	}
 }
 
-func getVersions() ([]*github.RepositoryRelease, error) {
-	client := githubutil.Client()
+func PrintUpStreamVersions(output io.Writer, minSupportedVersion string) {
+	resp, err := getResponseBody("https://registry.hub.docker.com/v2/repositories/openshift/origin/tags/")
+	if err != nil {
+		fmt.Fprintf(output, "Error Occured %s", err)
+		return
+	}
+	defer resp.Body.Close()
+	decoder := json.NewDecoder(resp.Body)
+	var data ImageTags
+	err = decoder.Decode(&data)
+	if err != nil {
+		fmt.Printf("%T\n%s\n%#v\n", err, err, err)
+		return
+	}
+	fmt.Fprintf(output, "The following OpenShift versions are available: \n")
+	for _, imageinfo := range data.Results {
+		if versionOrdinal(imageinfo.Name) >= versionOrdinal(minSupportedVersion) {
+			fmt.Fprintf(output, "\t- %s\n", imageinfo.Name)
+		}
+	}
+}
 
-	releases, resp, err := client.Repositories.ListReleases(githubOwner, githubRepo, nil)
+func getResponseBody(url string) (resp *http.Response, err error) {
+	resp, err = http.Get(url)
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
+	return resp, nil
+}
 
-	if len(releases) == 0 {
-		return nil, errors.New("There are no OpenShift versions available.")
+func versionOrdinal(version string) string {
```

Comment:
@hferentschik our downstream release doesn't follow Semantic version (http://semver.org/) that's the reason I didn't use it.